api/src/main/java/org/openmrs/module/registrationcore/api/impl/PatientActionListener.java

```diff
@@ -71,12 +125,45 @@ protected String getMessagePropertyValue(Message message, String propertyName) {
         }
     }
 
+    /**
+     * Prepares the parameters required for sending the patient to the MPI (the message)
+     * via the error handling module (specified in the global properties)
+     * in the case of initial failure.
+     *
+     * @param patient patient to be sent to MPI
+     * @return parameters as a string
+     */
+    protected String prepareParameters(Patient patient) {
+        SendingPatientToMpiParameters parameters = new SendingPatientToMpiParameters(patient.getUuid());
+        try {
+            return new ObjectMapper().writeValueAsString(parameters);
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot prepare parameters for OutgoingMessageException", e);
+        }
+    }
+
+    /**
+     * Validate that the message is an instance of MapMessage.
+     *
+     * @param message message to be validated
+     */
     private void validateMessage(Message message) {
-        if (!(message instanceof MapMessage))
+        if (!(message instanceof MapMessage)){
             throw new APIException("Event message should be MapMessage, but it isn't");
+        }
     }
 
+    /**
+     * Gets patient from local DB using patient uuid provided.
+     *
+     * @param patientUuid uuid of patient to be retrieved.
+     * @return retrieved patient
+     */
     private Patient getPatient(String patientUuid) {
-        return patientService.getPatientByUuid(patientUuid);
+        Patient patient = patientService.getPatientByUuid(patientUuid);
+        if (patient == null){
+            log.error("Unable to retrieve patient by uuid");
```

Comment:
Shouldn't this fail with an exception?

Reply:
swapped to 
`throw new RuntimeException("Unable to retrieve patient by uuid");`